graviti/dataframe/frame.py

```diff
@@ -126,7 +126,26 @@ def __getitem__(self, key: Union[str, Iterable[str]]) -> Container:
         raise KeyError(key)
 
     def __setitem__(self, key: str, value: Iterable[Any]) -> None:
-        pass
+        # self.schema.to_builtin always returns record.
+        field = self.schema.to_builtin().fields[key]  # type: ignore[attr-defined]
+        if isinstance(value, Container):
+            if not value.schema.to_pyarrow().equals(field.to_pyarrow()):
+                raise TypeError("The schema is mismatched with the given value.")
```

Comment:
I think is acceptable to replace the column with another column with different schema